docs/getting-started-guides/openstack-heat.md

```diff
@@ -0,0 +1,239 @@
+export OS_USERNAME=username
+export OS_PASSWORD=password
+export OS_TENANT_NAME=projectName
+export OS_AUTH_URL=https://identityHost:portNumber/v2.0
+export OS_TENANT_ID=tenantIDString
+export OS_REGION_NAME=regionName
+```
+
+#### Set additional configuration values
+
+In addition, here are some commonly changed variables specific to this provider, with example values. Under most circumstances you will not have to change these. Please see the files in the next section for a full list of options.
+
+```sh
+export STACK_NAME=KubernetesStack
+export NUMBER_OF_MINIONS=3
+export MAX_NUMBER_OF_MINIONS=3
+export MASTER_FLAVOR=m1.small
+export MINION_FLAVOR=m1.small
+export EXTERNAL_NETWORK=public
+export DNS_SERVER=8.8.8.8
+export IMAGE_URL_PATH=http://cloud.centos.org/centos/7/images
+export IMAGE_FILE=CentOS-7-x86_64-GenericCloud-1510.qcow2
+export SWIFT_SERVER_URL=http://192.168.123.100:8080
+export ENABLE_PROXY=false
+```
+#### Manually overriding configuration values
+
+If you do not have your environment variables set, or do not want them consumed, modify the variables in the following files under `cluster/openstack-heat`:
+
+- **[config-default.sh](http://releases.k8s.io/{{page.githubbranch}}/cluster/openstack-heat/config-default.sh)** Sets all parameters needed for heat template.
+- **[config-image.sh](http://releases.k8s.io/{{page.githubbranch}}/cluster/openstack-heat/config-image.sh)** Sets parameters needed to download and create new OpenStack image via glance.
+- **[openrc-default.sh](http://releases.k8s.io/{{page.githubbranch}}/cluster/openstack-heat/openrc-default.sh)** Sets environment variables for communicating to OpenStack. These are consumed by the cli tools (heat, glance, swift, nova).
+- **[openrc-swift.sh](http://releases.k8s.io/{{page.githubbranch}}/cluster/openstack-heat/openrc-swift.sh)** Some OpenStack setups require the use of seperate swift credentials. Put those credentials in this file.
+
+Please see the contents of these files for documentation regarding each variable's function.
+
+## Starting a cluster
+
+Once you've installed the OpenStack CLI tools and have set your OpenStack environment variables, issue this command:
+
+```sh
+export KUBERNETES_PROVIDER=openstack-heat; curl -sS https://get.k8s.io | bash
+```
+Alternatively, you can download a [Kubernetes release](https://github.com/kubernetes/kubernetes/releases) and extract the archive. To start your cluster, open a shell and run:
+
+```sh
+cd kubernetes # Or whichever path you have extracted the release to
+KUBERNETES_PROVIDER=openstack-heat ./cluster/kube-up.sh
+```
+Or, if you are working from a checkout of the Kubernetes code base, and want to build/test from source:
+
+```sh
+cd kubernetes # Or whatever your checkout root directory is called
+make clean
+make quick-release
+KUBERNETES_PROVIDER=openstack-heat ./cluster/kube-up.sh
+```
+## Inspect your cluster
+
+Once kube-up is finished, your cluster should be running:
+
+```console
+./cluster/kubectl.sh get cs
```

Comment:
Regarding the kubectl.sh, I've subscribed to this: https://github.com/kubernetes/kubernetes/issues/8817

In terms of user experience, kubectl.sh provides the easiest way to start with k8s, especially from the get.k8s.io perspective (`curl | bash`). 

Once kubectl.sh has been converted to an installer for kubectl, I'll update the docs accordingly.